### ISSUE TITLE
what does embedding_size mean

### ISSUE THREAD

**Issue by sandysdy** [KevinMusgrave/pytorch-metric-learning#106]
Hello！How can I understand "embedding_size" in the codes? 
Would you please give me an example to explain?

**Comment by KevinMusgrave** [comment]
It's the size of the vectors that your model outputs. In metric-learning, your model outputs a vector for each sample. You train the model so that similar samples are close together in that vector space, and dissimilar samples are far apart in that vector space.

**Comment by sandysdy** [comment]
Thanks! But I want to further verify my understanding.
I'm in an  images binary classification issure. The batchsize is 128, thus my model outputs a tensor of size([128,2]). So my embedding_size should be 2. Is that correct?

**Comment by KevinMusgrave** [comment]
In your case, the embedding size would be the size of the 2nd last layer. For example, let's say you are using a simple fully connected network with input size 1000, and output size 2:

```1000 -> 500 -> 2```

Because you are using the output for classification, then in the context of this library, ```embedding_size``` refers to the size of the 2nd last layer, which is 500. Specifically, if you're using one of the classification losses (ArcFaceLoss, CosFaceLoss, NormalizedSoftmaxLoss etc), then you should set ```embedding_size``` to 500, and ```num_classes``` to 2. The loss function will contain fully connected layer that maps from the embedding space (size 500) to the binary classification result (size 2).

**Comment by sandysdy** [comment]
Thank you for your detailed answer.  I understand what you mean.
Maybe you also helped me find out the reason of inferior performance when I use the DML lose, 
I should pay more attention to the second last layer , but not the last one in a binary classification case.
Thank you very much!

**Comment by KevinMusgrave** [comment]
Great! I'll close this for now. Please re-open this issue if you have further questions.
